scripts/discourse-sync/client.py

```diff
@@ -0,0 +1,22 @@
+import pydiscourse
+
+# Client extends pydiscourse.DiscourseClient with new methods
+class Client(pydiscourse.DiscourseClient):
+    def lock_post(self, post_id, locked, **kwargs):
+        """
+        Lock a post from being edited
+        https://docs.discourse.org/#tag/Posts/operation/lockPost
+
+        Args:
+            post_id: integer
+            locked: bool
+        """
+        if bool(locked):
+            kwargs["locked"] = "true"
+        else:
+            kwargs["locked"] = "false"
+        return self._put("posts/{}/locked.json".format(post_id), **kwargs)
+
+    def add_staff_notice(self, post_id, notice, **kwargs):
```

Comment:
More nits, but if you did a docstring for the other one, might as well here :smile:

```suggestion
    def add_staff_notice(self, post_id: int, notice: str, **kwargs):
        """Add a notice to the post that will only display to staff."""
```